### DISCUSSION TITLE
Organizational Link - CreateOrganizationalSharingLinkAsync

### DISCUSSION THREAD

**Discussion by miguelrmarques** [pnp/pnpcore#867] [Q&A]
I'm trying to get a link for a file to share with the organization by using CreateOrganizationalSharingLinkAsync method.

I'm using a app registration that has the following permissions:

![image](https://user-images.githubusercontent.com/29754732/171508526-d0cc4919-0821-45e8-8243-520a16941d15.png)

I'm using the PnP.Core (1.6.47-nightly)
When I try to create a link it gives me an error of access denied
- Microsoft Graph service exception /  PnP.Core.ErrorType.GraphServiceError / Access Denied

![image](https://user-images.githubusercontent.com/29754732/171509376-be230d90-cb23-4873-b479-e309879decce.png)

And also in the sharepoint admin I have the following policy

![image](https://user-images.githubusercontent.com/29754732/171509893-ef8eb12a-a054-4943-909c-f62ca3cc1204.png)

Any idea?
Thanks in advance.

**Comment by MathijsVerbeeck** [comment]
Hi @miguelrmarques 

I've just tested this locally with the exact same permissions in my App Registration and it is working just fine:

![image](https://user-images.githubusercontent.com/9193870/171576827-e01f0d53-1200-4f1c-ac63-dd6d6b9de51d.png)

Could you, if possible, send me a Fiddler trace of the request, so I can verify the access token that you're using and ensure that all the permissions are obtained correctly?
